### DISCUSSION TITLE
Does OSQP support SOCP or QCQP?

### DISCUSSION THREAD

**Discussion by MizWong** [osqp/osqp#602] [General]
From the examples and descriptions, seems only linear constraints are possible?

**Comment by imciner2** [comment]
Hi @MizWong, sorry for no one replying to your question earlier. No, the OSQP solver as implmented does not solve SOCP or QCQP optimization problems. OSQP supports only linear constraints of the form $l \leq Ax \leq u$. (That said, the ADMM algorithm could be extended to other problem types given the proper theoretical/implementation details are worked out).

If you need SOCP problems, then another solver would be needed, such as Clarabel (https://github.com/oxfordcontrol/Clarabel.cpp, or any of the other language interfaces it provides).

For QCQPs, I know that HPIPM (https://github.com/giaf/hpipm) does provide support for some forms, so you can look into that solver should you need them.
